src/pages/DetailsPage.js

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from "react";
+
+import Badge from "../components/Badge";
+import Header from "../components/Header";
+import Icon from "./../components/Icon";
+
+const DetailsPage = (props) => {
+	const pokemonId = props.match.params.pokemonId;
+	const [pokemon, setPokemon] = useState({});
+	const getPokemonById = async (pokemonId) => {
+		let pokemon = {};
+		try {
+			pokemon = await fetch(
+				`https://pokeapi.co/api/v2/pokemon/${pokemonId}`
+			).then((response) => response.json());
+		} catch (error) {
+			console.log(error);
+		}
+		console.log(pokemon);
+		return pokemon;
+	};
+	useEffect(async () => {
```

Comment:
A useEffect callback should not be async. Instead you can call inside an async function (but you can't await for it).